utils.go

```diff
@@ -296,7 +296,10 @@ func ToObjectInfo(bucketName string, objectName string, h http.Header) (ObjectIn
 
 	// extract lifecycle expiry date and rule ID
 	expTime, ruleID := amzExpirationToExpiryDateRuleID(h.Get(amzExpiration))
-
+	deleteMarker := false
+	if strings.ToLower(h.Get(amzDeleteMarker)) == "true" {
+		deleteMarker = true
+	}
```

Comment:
Same as above comment 

- no need to use strings.ToLower()
- one line `deleteMarker := h.Get(amzDeleteMarker) == "true"`